services/addFilter.js

```diff
@@ -0,0 +1,20 @@
+export default function addFilter(array, filter) {
+  const date = new Date();
+  const time = date.getHours();
+  const month = date.getMonth() + 1;
+
+  const filterSpecies = el => !filter?.species || el.species === filter.species;
+  const filterPersonality = el =>
+    !filter?.personality || el.personality === filter.personality;
+  const filterShadow = el => !filter?.shadow || el.shadow === filter.shadow;
+  const availability = el =>
+    !filter?.isAvailable ||
+    ((el.northernMonthsArray.includes(month) || el.isAllYear) &&
+      (el.timeArray.includes(time) || el.isAllDay));
+
+  return array
+    .filter(filterPersonality)
+    .filter(filterSpecies)
+    .filter(filterShadow)
+    .filter(availability);
```

Comment:
```suggestion
  const keyFilter = (el, key) => el[key] === !filter[key] ? el[key] : filter[key];

  return array
    .filter(el => keyFilter(el, 'species')
    .filter(el => keyFilter(el, 'personality')
    .filter(el => keyFilter(el, 'shadow')
    .filter(availability);
```

this might also be another solution, not as easy to read, but if there are more filters coming it definitly helps. Just wanted to show you another way